website/docs/advanced-concepts/access-control/granular-access-control/roles.md

```diff
@@ -0,0 +1,242 @@
+
+#### Create
+
+**Create** gives the ability to create datasources in a workspace. In Datasource and queries, you can give the create permission at different hierarchical levels -
+
+##### Workspace
+
+If you give create permission to the workspace, it gives access to create datasources in the respective workspace. 
+
+##### Datasource
+
+If you want the user to have create access only to a specific datasource in a workspace, open the respective workspace tree and check the create permission adjacent to the respective datasource. **Create** for a particular datasource allows the user to make tweaks the datasource configurations and further test, delete or save it.
+You can't create queries for a datasource by only giving create access for the datasources in the workspace. To create new queries, check the create permission for the datasource and give the create access to queries under [Application resources permissions](#application-resources-permissions) 
+
+#### Edit 
+
+**Edit** allows the user to change/modify the existing datasources. You can edit the datasource name, configurations and further test and save the datasource. Edit permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give edit permission to the workspace, it gives access to modify all the datasources in the respective workspace. 
+
+##### Datasource
+
+If you want the user to have edit access only to a specific datasource in a workspace, open the respective workspace tree and check the edit permission adjacent to the respective datasource. **Edit** for just a datasource allows you to modify the name, configurations of that datasource.
+
+#### View
+
+**View** only allows a user to have view access for the existing datasources. The user can interact with the functionality of the datasource, test the datasource, but can't make any changes with View permission. View permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give view permission to the workspace, it gives access to view and execute all the datasources in the respective workspace.
+
+##### Datasource 
+
+If you want the user to have view access only to a specific datasource in a workspace, open the respective workspace tree and check the view permission adjacent to the respective datasource. **View** for just a datasource allows you to view, test and execute that datasource.
+
+#### Delete
+
+**Delete** gives the user an access to delete the existing datasources in a workspace. With the delete permission, one can't edit or create a datasource, but, the user has the access to delete it.
+Delete permission can be given at various hierarchical levels - 
+
+##### Workspace
+
+If you give delete permission to the workspace, It gives access to remove any datasource from the respective workspace. 
```

Comment:
unless the datasource is being used in any existing queries.

Reply:
If a datasource is used in any existing queries, can it not be deleted? Even if a user has delete access for it?

Reply:
nope... it will give you error on the application edit screen that this datasource is being used by a few queries.